### DISCUSSION TITLE
How to use oas-client.json

### DISCUSSION THREAD

**Discussion by Lunr127** [lucaro/DRES#379] [Q&A] [answered]
Hi, there
You wirte it in Interaction Logging: "We strongly recommend the usage of the [client OpenApi Specification] to generate the code to submit logging information (and generally interact with the server)!"
But I'm not sure how to use it. Should I adjust the contents of this json file, oas-client.json, and upload the entire file to the server?
In Client-Examples, for example, Client.java, the method of submission is as follows
```
try {
            submissionResponse = submissionApi.getApiV1Submit(
                    null, //does not usually need to be set
                    "some_item_name", //item which is to be submitted
                    null, //in case the task is not targeting a particular content object but plaintext
                    null, // for items with temporal components, such as video
                    null,  // only one of the time fields needs to be set.
                    "00:00:10:00", //in this case, we use the timestamp in the form HH:MM:SS:FF
                    sessionId
            );
        } 
```
How should I modify the code above to directly submit the json file to the server?
Thanks for answering！

**Comment by lucaro** [comment, marked as answer]
I'm not sure I understand the question. The json file you mention contains the [OpenAPI](https://en.wikipedia.org/wiki/OpenAPI_Specification) specification of the API, its endpoints, parameters, and response messages. The Java example is only that, an example, on how it would look like if you would talk to this API using Java, here with automatically generated client code based on the specification. You can however use pretty much any language you want, as there are OpenAPI code generators for a large range of languages. Just use the code generator to generate client code based on the specification in the json and use that code to communicate with the server.

**Reply by Lunr127** [reply]
Thanks very much for your explanation. Yes, I misunderstood the purpose of oas-client.json.
But I'm sorry, I still have some doubts about how to submit the results to the server. I am a member of the VBS competition team this year. Because it is the first time to participate in the competition, I have encountered some difficulties in submitting the results to the server. I'm going to submit the results in Java. Now I have successfully logged in and displayed the VBS test questions like this
```
Found 1 ongoing evaluation runs
VBS2023 Test (30ecf0ac-b1b4-41e4-96fe-20ac6768e25d): ACTIVE
Test tasks for VBS 2023
```
When I want to submit the results to the designated questions, should I modify the "some_item_name" and else in the following code:
```
try {
            submissionResponse = submissionApi.getApiV1Submit(
                    null, //does not usually need to be set
                    "some_item_name", //item which is to be submitted
                    null, //in case the task is not targeting a particular content object but plaintext
                    null, // for items with temporal components, such as video
                    null,  // only one of the time fields needs to be set.
                    "00:00:10:00", //in this case, we use the timestamp in the form HH:MM:SS:FF
                    sessionId
            );
        } 
```
I modified the code above, change "some_item_name" to "30ecf0ac-b1b4-41e4-96fe-20ac6768e25d", but the submission failed:
`There is currently no active task which would accept submissions.`
In addition, I want to know whether multiple answers can be submitted at one time, for example, several answers can be submitted to the DERS system in the form of JSON files.
I would appreciate it if you could take the time to tell me how to submit the answers to the DERS system of the VBS competition using Java programs. I'm very sorry to have taken up your time.

**Reply by lucaro** [reply]
The item name you are supposed to submit is not an id but the actual name of the video in this case. The API specifies currently only single submissions, so you need to submit your results one at a time. The OpenAPI json file is only for the specification of the API, it is never to be sent to the server in any way.

**Reply by Lunr127** [reply]
It works! Thanks a lot!
